### ISSUE TITLE
Opening multiple dialogs simultaneously does not focus the top dialog

### ISSUE THREAD

**Issue by JannicBeck** [angular/components#8046]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When multiple dialogs are open the focus should transfer to the next dialog when closing the first one.

#### What is the current behavior?
The next dialog is not in focus when closing one dialog.

#### What are the steps to reproduce?
Providing a StackBlitz/Plunker (or similar) is the *best* way to get the team to see your issue. <br/>
Plunker starter (using on `@master`): https://goo.gl/DlHd6U<br/>
StackBlitz starter (using latest `npm` release): https://goo.gl/wwnhMV<br/>
http://plnkr.co/edit/L3mmNpTBRKnWL5Edn9SX?p=preview
Press button Open two dialogs, then press enter repeatedly.
It should close both dialogs, but instead dialog second dialog is closed on first enter and after that first dialog is not closed but the Open two dialogs button is focused and two new dialogs are opened on pressing enter.

#### What is the use-case or motivation for changing an existing behavior?
To be able to close all dialogs with keypress and not focus stuff that is behind the dialog.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
All material versions

**Comment by merlosy** [comment]
Have you tried opening the second modal from the first one?

AFAIK, opening multiple modal at the same time is bad UX, even if I like lasagna :p

**Comment by JannicBeck** [comment]
Its just an example to reproduce. Imagine observables triggering the dialogs ...

**Comment by crisbeto** [comment]
It doesn't work because you're opening them at the same time. Dialogs move focus after the animation is finished.

**Comment by jelbourn** [comment]
@crisbeto will your PR to perform focus immediately make it always focus the last opened dialog as a side-effect?

**Comment by crisbeto** [comment]
I'm not sure that it would solve the problem entirely. Focus should be moved between the dialogs, but it won't be transferred to the first focusable element since we still need to wait for the content to initialize.

**Comment by crisbeto** [comment]
This should've been resolved by https://github.com/angular/components/pull/23085.
